bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/QbusBridgeHandler.java

```diff
@@ -0,0 +1,360 @@
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    /**
+     * Schedule future communication refresh.
+     *
+     * @param interval_config Time before refresh in minutes.
+     */
+
+    private void setupRefreshTimer(int refreshInterval) {
+        ScheduledFuture<?> timer = refreshTimer;
+        if (timer != null) {
+            timer.cancel(true);
+            refreshTimer = null;
+        }
+
+        if (refreshInterval == 0) {
+            return;
+        }
+
+        // This timer will check connection with server and client periodically
+        refreshTimer = scheduler.scheduleWithFixedDelay(() -> {
+            QbusCommunication comm = getCommunication();
+
+            if (comm != null) {
+                if (!comm.communicationActive()) {
+                    // Disconnected from Qbus Server, try to reconnect
+                    try {
+                        comm.restartCommunication();
+                    } catch (InterruptedException e) {
+                        bridgeOffline("No connection with Qbus Server" + e.toString());
+                    } catch (IOException e) {
+                        bridgeOffline("No connection with Qbus Server" + e.toString());
+                    }
+                    if (!comm.communicationActive()) {
+                        bridgeOffline("No connection with Qbus Server");
+                        return;
+                    }
+                } else {
+                    // Controller disconnected from Qbus client, try to reconnect controller
+                    if (!comm.clientConnected()) {
+                        try {
+                            comm.restartCommunication();
+                        } catch (InterruptedException e) {
+                            bridgeOffline("No connection with Qbus Server" + e.toString());
+                        } catch (IOException e) {
+                            bridgeOffline("No connection with Qbus Server" + e.toString());
+                        }
+                        if (!comm.clientConnected()) {
+                            bridgeOffline("No connection with Qbus Client");
+                            return;
+                        }
+                    }
+                }
+            }
+            updateStatus(ThingStatus.ONLINE);
+
+        }, refreshInterval, refreshInterval, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Disposes the Bridge and stops communication with the Qbus server
+     */
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> timer = refreshTimer;
+        if (timer != null) {
+            timer.cancel(true);
+        }
+
+        refreshTimer = null;
+
+        QbusCommunication comm = getCommunication();
+
+        if (comm != null) {
+            try {
+                comm.stopCommunication();
+            } catch (IOException e) {
+                String message = e.toString();
+                logger.error("Error on stopping communication.{} ", message);
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be debug. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging